melodies_monet/driver.py

```diff
@@ -619,14 +732,106 @@ def open_model_files(self, time_interval=None, control_dict=None):
 
         else:
             print('**** Reading Unspecified model output. Take Caution...')
+            from monetio import models
             if len(self.files) > 1:
                 self.obj = xr.open_mfdataset(self.files,**self.mod_kwargs)
             else:
                 self.obj = xr.open_dataset(self.files[0],**self.mod_kwargs)
+                
         self.mask_and_scale()
         self.rename_vars() # rename any variables as necessary 
         self.sum_variables()
 
+        if self.extra_calc is not None:
+            print("Performing extra model calculations...")
+            
+            if "dewpoint" in self.extra_calc:
+                print("Calculating modeled Dewpoint...")
+                from .util.metcalc import dewpoint # import functions from the util.metcalc file
+                
+                varmap = self.extra_calc["dewpoint"]
+                self.obj=dewpoint(self.obj, varmap = varmap)
+                #print(self.obj)
+
+            if "rel_hum" in self.extra_calc:
+                print("Calculating modeled relative humidity...")
+                from .util.metcalc import relh
+
+                varmap = self.extra_calc["rel_hum"]
+                self.obj=relh(self.obj, varmap = varmap)
+                #print(self.obj)
+
+            if "windspeed" in self.extra_calc:
+                print("Calculating modeled windpseed...")
+                from .util.metcalc import wspd
+
+                varmap = self.extra_calc["windspeed"]
+                self.obj=wspd(self.obj, varmap = varmap)
+                #print(self.obj)
+                
+            if "winddir" in self.extra_calc:
+                print("Calculating modeled wind direction...")
+                from .util.metcalc import wdir
+
+                varmap = self.extra_calc["winddir"]
+                self.obj=wdir(self.obj, varmap = varmap)
+                #print(self.obj)   
+
+            if "wind_barb" in self.extra_calc:
+                print("Calculating model wind barbs...")
+                u_comp = self.extra_calc.get('wind_barb', {}).get("u_comp", None)
+                v_comp = self.extra_calc.get('wind_barb', {}).get("v_comp", None)         
+            
+            if "ptemp_mod" in self.extra_calc:
+                print("Calculating modeled potential temperature...")
+                from .util.metcalc import ptemp
+            
+                varmap = self.extra_calc["ptemp_mod"]
+
+                # Extract a single vertical column from the model field 
+                # for boundary layer calc
+                pres = self.obj[varmap.get("pres", "pressure_model")]
+                temp = self.obj[varmap.get("temp", "temperature_k")]
+
```

Comment:
In your YAML file example, you use temperature here instead of temp? Should this be temperature instead?
https://github.com/L12D2/MELODIES-MONET/blob/BoundaryLayerHeight_Tropopause_calc_beta/docs/examples/control_read_looped_aircraft_UWY_UFS_CHEM.yaml